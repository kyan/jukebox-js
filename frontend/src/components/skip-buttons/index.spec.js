import React from 'react'
import { mount } from 'enzyme'
import SkipButtons from './index'

describe('SkipButtons', () => {
  const prevMock = jest.fn()
  const nextMock = jest.fn()
  let wrapper

  describe('render', () => {
    wrapper = mount(<SkipButtons onPrevious={prevMock} onNext={nextMock} />)

    it('renders the as expected', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('handles a prev click', () => {
      prevMock.mockClear()
      wrapper.find('PreviousButton').simulate('click')
      expect(prevMock).toHaveBeenCalled()
    })

    it('handles a next click', () => {
      nextMock.mockClear()
      wrapper.find('NextButton').simulate('click')
      expect(nextMock).toHaveBeenCalled()
    })
  })
})
