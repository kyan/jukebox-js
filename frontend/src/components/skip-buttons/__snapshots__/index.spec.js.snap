// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkipButtons render renders the as expected 1`] = `
<SkipButtons
  onNext={[MockFunction]}
  onPrevious={[MockFunction]}
>
  <PreviousButton
    onClick={[MockFunction]}
  >
    <Button
      animated="vertical"
      as="button"
      className="jb-previous-button"
      onClick={[MockFunction]}
    >
      <Ref
        innerRef={
          Object {
            "current": <button
              class="ui vertical animated button jb-previous-button"
            >
              <div
                class="hidden content"
              >
                Prev
              </div>
              <div
                class="visible content"
              >
                <i
                  aria-hidden="true"
                  class="step backward icon"
                />
              </div>
            </button>,
          }
        }
      >
        <RefFindNode
          innerRef={
            Object {
              "current": <button
                class="ui vertical animated button jb-previous-button"
              >
                <div
                  class="hidden content"
                >
                  Prev
                </div>
                <div
                  class="visible content"
                >
                  <i
                    aria-hidden="true"
                    class="step backward icon"
                  />
                </div>
              </button>,
            }
          }
        >
          <button
            className="ui vertical animated button jb-previous-button"
            onClick={[Function]}
          >
            <ButtonContent
              hidden={true}
            >
              <div
                className="hidden content"
              >
                Prev
              </div>
            </ButtonContent>
            <ButtonContent
              visible={true}
            >
              <div
                className="visible content"
              >
                <Icon
                  as="i"
                  name="step backward"
                >
                  <i
                    aria-hidden="true"
                    className="step backward icon"
                    onClick={[Function]}
                  />
                </Icon>
              </div>
            </ButtonContent>
          </button>
        </RefFindNode>
      </Ref>
    </Button>
  </PreviousButton>
  <NextButton
    onClick={[MockFunction]}
  >
    <Button
      animated="vertical"
      as="button"
      className="jb-next-button"
      onClick={[MockFunction]}
    >
      <Ref
        innerRef={
          Object {
            "current": <button
              class="ui vertical animated button jb-next-button"
            >
              <div
                class="hidden content"
              >
                Next
              </div>
              <div
                class="visible content"
              >
                <i
                  aria-hidden="true"
                  class="step forward icon"
                />
              </div>
            </button>,
          }
        }
      >
        <RefFindNode
          innerRef={
            Object {
              "current": <button
                class="ui vertical animated button jb-next-button"
              >
                <div
                  class="hidden content"
                >
                  Next
                </div>
                <div
                  class="visible content"
                >
                  <i
                    aria-hidden="true"
                    class="step forward icon"
                  />
                </div>
              </button>,
            }
          }
        >
          <button
            className="ui vertical animated button jb-next-button"
            onClick={[Function]}
          >
            <ButtonContent
              hidden={true}
            >
              <div
                className="hidden content"
              >
                Next
              </div>
            </ButtonContent>
            <ButtonContent
              visible={true}
            >
              <div
                className="visible content"
              >
                <Icon
                  as="i"
                  name="step forward"
                >
                  <i
                    aria-hidden="true"
                    className="step forward icon"
                    onClick={[Function]}
                  />
                </Icon>
              </div>
            </ButtonContent>
          </button>
        </RefFindNode>
      </Ref>
    </Button>
  </NextButton>
</SkipButtons>
`;
